### DISCUSSION TITLE
Airflow 2.3.2 grid view failing for some DAGs due to ui_fgcolor error

### DISCUSSION THREAD

**Discussion by ayetealab** [apache/airflow#25927] [General]
Following an upgrade from 2.1.0 to 2.3.2 when I open the grid view for some DAGs (not all), it does not render the main grid area in the same way as described here: https://github.com/apache/airflow/discussions/23908. The fix for that issue was part of 2.3.2 which I am already running so the symptom is the same but the root cause differs.

In the Airflow web server logs I see errors like this:

```
Aug 23 15:30:55 airflow.my-domain.org airflow-webserver.sh[5762]: [2022-08-23 15:30:55,125] {validation.py:244} ERROR - https://airflow.my-domain.org/api/v1/dags/my-dag-id/tasks validation error: 'black' does not match '^#[a-fA-F0-9]{3,6}$'
Aug 23 15:30:55 airflow.my-domain.org airflow-webserver.sh[5762]: Failed validating 'pattern' in schema['properties']['tasks']['items']['properties']['ui_fgcolor']:
Aug 23 15:30:55 airflow.my-domain.org airflow-webserver.sh[5762]: {'description': 'Color in hexadecimal notation.',
Aug 23 15:30:55 airflow.my-domain.org airflow-webserver.sh[5762]: 'pattern': '^#[a-fA-F0-9]{3,6}$',
Aug 23 15:30:55 airflow.my-domain.org airflow-webserver.sh[5762]: 'type': 'string'}
Aug 23 15:30:55 airflow.my-domain.org airflow-webserver.sh[5762]: On instance['tasks'][4]['ui_fgcolor']:
Aug 23 15:30:55 airflow.my-domain.org airflow-webserver.sh[5762]: 'black'
```

If I edit this [api connexion line](https://github.com/apache/airflow/blob/0bf5f495d4131109fba449697adee68a62516851/airflow/api_connexion/openapi/v1.yaml#L3653 ) on the Airflow host from `pattern: ^#[a-fA-F0-9]{3,6}$` to `pattern: ^#[a-fA-F0-9]{3,6}$|black` then my problem goes away - the grid view renders perfectly for all DAGs. It's a workaround but not an ideal one. 

I have hunted everywhere and I can not fathom where ui_fgcolor = black instead of a HEX value is being sourced from for _some_ DAGs. I also can not find any consistency between which DAGs are effected by this. Does anyone have any suggestions?

**Comment by potiuk** [comment]
Is it still a problem in 2.3.4?

**Reply by potiuk** [reply]
Upgrade to 2.3.4 is inevitable, so I recommend to do so.
